agent/consul/state/catalog.go

```diff
@@ -676,6 +676,9 @@ func (s *Store) deleteNodeTxn(tx WriteTxn, idx uint64, nodeName string, entMeta
 	if err := catalogUpdateNodesIndexes(tx, idx, entMeta); err != nil {
 		return fmt.Errorf("failed updating index: %s", err)
 	}
+	if err := catalogUpdateNodeIndexes(tx, nodeName, idx, entMeta); err != nil {
+		return fmt.Errorf("failed updating node index: %s", err)
+	}
```

Comment:
I think for deletes we probably need to do something like `catalogUpdateServiceExtinctionIndex`. If we track an index for every deleted node, that will grow forever. We'll keep an index for every deleted node name, which is unbounded.

I think by using an extinction index for deleted items we only ever have 1 entry for every deleted node. It's slightly less optimal in terms of reducing blockingQuery churn, but it is much safer in terms of space used by the index tracking.

Reply:
Yes, I noticed that extra 'extinction index' used by the `services` query implementation for this purpose, and punted on it here since it didn't affect my use-case (the nodes in our cluster are pretty stable), I agree it's necessary in general though. I can work on adding it sometime soon.

Reply:
OK, my first attempt at this is in 440db29cb1e7326e1abbbb6239c6cbb64718d7bb, but it's largely a copy-paste of `catalogUpdateServiceExtinctionIndex` and related methods (this code is all unfamiliar territory to me). I haven't tested beyond the existing unit tests, and could use some guidance on any extra testing/refactoring that might be helpful for this extra part.